### DISCUSSION TITLE
Why onMutate isn't firing immediately?

### DISCUSSION THREAD

**Discussion by evrrnv** [TanStack/query#7382] [Q&A]
When I call`setQueryData` before calling the mutation, it updates faster than calling `setQueryData` inside `onMutate` in react-query react-native, can anybody explains to me why this is happening?

https://github.com/TanStack/query/assets/75393305/a32d7dc8-425e-4f91-a008-c566f4803c17

https://github.com/TanStack/query/assets/75393305/0adaec12-296c-4c90-a37b-8f471ab68d9b

You can notice the update is happing immediately in the first screen record when calling `setQueryData` before calling the mutation, but in the second screen record, I'm calling setQueryData inside `onMutate`

**Comment by riley-paul** [comment]
Did you determine what was causing this? I am noticing the same thing.

**Reply by evrrnv** [reply]
The issue was related to my code, not related to react-query
